dask/dataframe/io/csv.py

```diff
@@ -688,7 +695,8 @@ def to_csv(
     nanRep : None
         deprecated, use na_rep
     mode : str
-        Python write mode, default 'w'
+        Python write mode, default 'w'. Note that not all filesystems
+        support the append mode.
```

Comment:
Is this only important in the single file mode?  If so, I recommend that we put this note under `single_file`.  This is unlikely to be relevant to users in the many csv files case, which I expect to be more common.

Reply:
Also, are there file systems where this is known to fail or provide inaccurate results?  For example what about eventually consistent cloud storage systems like S3/GCS?  Should we err quickly if the user tries to do something like the following? 

```
df.to_csv("s3://...", single_file=True)
```

Reply:
I have moved the note to the single_file entry.

I actually do not know how to deal with the S3/GCS case, and I am not familiar with the consistency models of these systems. A list of incompatible file systems may be hard to curate and change from time to time. My expectation is to follow the spirit of Python and trust the users. I have added a note regarding the network storage systems. An appropriate warning might also be good. I feel an error might be too strict. After all, if a user have some weird settings in her or his custom S3 storage server (Minio, Apache Oxide, or maybe others) and does want the behaviour, it would be unwise for us to prevent that.

Let me know of your opinion.